recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -49,8 +49,29 @@ this.NormandyApi = {
     return this.apiCall("post", endpoint, data);
   },
 
+  apiIndex() {
+    if (!indexPromise) {
+      indexPromise = this.get(prefs.getCharPref("api_url")).then(res => res.json);
+    }
+    return indexPromise;
+  },
+
+  makeApiUrl(endpoint) {
+    let base = prefs.getCharPref("api_url");
+    // Remove trailing slash if present
+    if (base.slice(-1) === "/") {
+      base = base.slice(0, -1);
+    }
+    // Remove leading slash, if present
+    if (endpoint[0] === "/") {
+      endpoint = endpoint.slice(1);
+    }
+    return `${base}/${endpoint}`;
+  },
+
   fetchRecipes: Task.async(function* (filters = {}) {
-    const recipeResponse = yield this.get("recipe/signed/", filters);
+    const signedRecipeUrls = (yield this.apiIndex())["recipes-signed"];
+    const recipeResponse = yield this.get(signedRecipeUrls, filters, {enabled: true});
```

Comment:
OK. For my second paragraph, I mostly meant that this line: https://github.com/mythmon/normandy/blob/b3da82e51da205d0deea69eaa7024170eaba2e51/recipe-server/normandy/base/tests/test_api.py#L18 seems to use `recipe-signed` and this line: https://github.com/mythmon/normandy/blob/b3da82e51da205d0deea69eaa7024170eaba2e51/recipe-client-addon/lib/NormandyApi.jsm#L73 seems to read out `recipeS-signed` (note extra s) for the key in a JSON structure. Now, I haven't checked the rest of the code in detail, so maybe I'm missing something and/or the `recipeS-signed` key (with lowercase s...) already exists pre-patch (... elsewhere?). But it looked odd, is all. :-)